### PR TITLE
Add basic framework support info datastructure

### PR DESCRIPTION
**Pull request by tmeasday**
Issue: Add info for framework support table / callouts in 6.0 docs

### CODE REVIEW

**Comment by kylesuss** [comment]
This seems good, but I might let it sit for a bit until I can do the styling on the framework support. Is this where you planned on stopping w/ this task @tmeasday ? Just want to clarify what else is left.

**Comment by tmeasday** [comment]
@kylesuss I think it depends on what we decide regarding the questions I had in the other PR. If we want to try and parse this information out of the tables, as @shilman suggested (I'm also kind of leaning that way), then I'm happy to do that work.

**Comment by tmeasday** [comment]
Here's what the page looks like now with the table generated from the data:

image

File: docs/frameworks.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    },
+    {
+      name: 'Storyshots',
+      unsupported: ['ember', 'mithril', 'marko'],
+      path: 'workflows/unit-testing',
+    },
+    {
+      name: 'storysource',
+      unsupported: [],
+      path: 'essentials/actions',
```
Comment (shilman) [inline docs/frameworks.js:71]: typo but not sure where it should link

File: docs/frameworks.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    },
+    {
+      name: 'CSF Stories',
+      unsupported: [],
+      path: 'api/csf',
+    },
+    {
+      name: 'storiesOf stories',
+      unsupported: [],
+      repoPath: 'lib/core/ADVANCED.md',
```
Comment (shilman) [inline docs/frameworks.js:91]: does this need to be delineated somehow since it's linking into the repo (I assume)

File: docs/frameworks.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    },
+    {
+      name: 'Source',
+      unsupported: [],
+      path: '?',
```
Comment (shilman) [inline docs/frameworks.js:96]: storysource above?

File: docs/frameworks.js [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    },
+    {
+      name: 'Notes/Info',
+      unsupported: ['html', 'riot'],
```
Comment (shilman) [inline docs/frameworks.js:100]: Info is ONLY supported for React, Notes is supported for everything AFAIK.

We have a deprecated-addons repo for these.

File: docs/frameworks.js [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+      path: '?',
+    },
+    {
+      name: 'Args Table',
+      unsupported: ['react', 'vue', 'angular', 'html', 'ember', 'rax'],
```
Comment (shilman) [inline docs/frameworks.js:105]: Supported for everything. Automated for: react, vue, angular, ember, web-components

File: docs/frameworks.js [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+      path: 'writing-docs/doc-blocks#argstable',
+    },
+    {
+      name: 'Controls',
+      supported: ['react', 'vue', 'rax'],
```
Comment (shilman) [inline docs/frameworks.js:110]: Supported for everything. Automated for: react, vue, angular, ember, web-components

File: docs/frameworks.js [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+      path: 'essentials/controls',
+    },
+    {
+      name: 'Description',
+      unsupported: ['react', 'vue', 'angular', 'ember', 'rax'],
```
Comment (shilman) [inline docs/frameworks.js:115]: Supported for everything. Automated for: react, vue, angular, ember, web-components

File: docs/frameworks.js [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+      path: 'FIXME',
+    },
+    {
+      name: 'Inline stories',
+      supported: ['react', 'vue', 'rax'],
```
Comment (shilman) [inline docs/frameworks.js:120]: Supported for react, vue, web-components, html

File: docs/frameworks.js (hunk 2 above)
Comment (shilman) [reply docs/frameworks.js:91]: $monorepo?

**Review by shilman — CHANGES_REQUESTED** [review]
Looks fantastic @tmeasday 

Table can be organized into essentials, other addons, docs features

Frameworks should be linked to their package in the monorepo OR framework pages in the docs when we make framework-specific intro snippets.

File: docs/frameworks.js [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+      name: 'jest',
+      unsupported: [],
+    },
+    {
+      name: 'knobs',
+      unsupported: [],
+    },
+    {
+      name: 'links',
+      unsupported: ['marko'],
```
Comment (shilman) [inline docs/frameworks.js:53]: Looks suspicious to me. Was this copied from the existing table or did you make it up?

Comment (tmeasday) [reply docs/frameworks.js:53]: Copied from existing table

**Comment by tmeasday** [comment]
@shilman I am merging this to get frontpage building again, but please let's sort out your comments above.

**Pull request merged**